rpc/wrtc_signaling_server.go

```diff
@@ -331,6 +349,34 @@ func (srv *WebRTCSignalingServer) Answer(server webrtcpb.SignalingService_Answer
 	}
 	defer srv.clearAdditionalICEServers(hosts)
 
+	// If heartbeats allowed (indicated by answerer), start goroutine to send
+	// heartbeats.
+	//
+	// The answerer does not respond to heartbeats. The signaling server is only
+	// using heartbeats to ensure the answerer is reachable. If the answerer is
+	// down, the heartbeat will error in the heartbeating goroutine below, the
+	// stream's context will be canceled, and we will stop handling interactions
```

Comment:
Does "the stream's context will be canceled, and we will stop handling interactions" refer to the `ctx` used in line 380 `RecvOffer(ctx, hosts)`?

Reply:
It is indeed that context. More specifically, that `ctx` value is really `server.Context()` which is the context stored on the bidi stream passed into `Answer`.

Reply:
Ok, this makes a lot more sense, thanks for that explanation! Sorry for one (last?) ask: can we incorporate that information into the documentation? The context handling here deviates from common patterns (I think out of necessity), so it's not obvious that the line 365 `server.Send` getting an error implies the context used on line 380 for a blocking operation gets canceled. Knowing this, I can see your current documentation touching all the important pieces. I just think some more hand-holding/babying of the reader in this specific case will be helpful.

Reply:
For sure! Thanks for asking for more clarity here; some of this stuff is opaque, so I agree that it's important to document what's going on here. I added
```
// .... We stop handling interactions because the stream's                          
// context (`ctx` here and below) is used in the `RecvOffer` call below this                                               
// goroutine that waits for a caller to attempt to establish a connection.  
```

I only added that piece to the signaling server side, as I think that concept might be more difficult to explain from the answerer side, but hopefully future NetCoders will be able to see the relationship between the server and answerer heartbeat logic.

Reply:
> I only added that piece to the signaling server side

For sure. In hindsight, now knowing where the action is happening, I would have better placed my questions. I think duplicating some of the content as documentation is perfectly fine though.